Linking fails with `'ItaniumNodes.def' file not found`

### Description of the bug:


While working on reverse introp I've occasionally been running a full compile-link-run test to make sure the code is doing what I think it's doing. This stopped working after [Switch to a Bazel-based runtimes build, and add bootstrapping](https://github.com/carbon-language/carbon-lang/commit/27870892472cbc96324184fb64c2099f95d95f5e). I now get this error:
```
INFO: Invocation ID: c8ff09b8-b4de-4927-9f1f-cac0b719b1cf
Computing main repo mapping: 
Loading: 
Loading: 0 packages loaded
bazel: Entering directory `/usr/local/google/home/nicholasbishop/.cache/bazel/_bazel_nicholasbishop/7a0bf1e4d37f50d2dd7c7e67a3c8032e/execroot/_main/'
Analyzing: target //toolchain:toolchain (0 packages loaded, 0 targets configured)
Analyzing: target //toolchain:toolchain (0 packages loaded, 0 targets configured)

INFO: Analyzed target //toolchain:toolchain (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
bazel: Leaving directory `/usr/local/google/home/nicholasbishop/.cache/bazel/_bazel_nicholasbishop/7a0bf1e4d37f50d2dd7c7e67a3c8032e/execroot/_main/'
Target //toolchain:carbon up-to-date:
  bazel-bin/toolchain/carbon
INFO: Elapsed time: 0.427s, Critical Path: 0.08s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/toolchain/carbon <args omitted>
In file included from /usr/local/google/home/nicholasbishop/.cache/bazel/_bazel_nicholasbishop/7a0bf1e4d37f50d2dd7c7e67a3c8032e/execroot/_main/bazel-out/k8-fastbuild/bin/toolchain/carbon.runfiles/_main/toolchain/install/runtimes/libcxxabi/src/cxa_demangle.cpp:17:
/usr/local/google/home/nicholasbishop/.cache/bazel/_bazel_nicholasbishop/7a0bf1e4d37f50d2dd7c7e67a3c8032e/execroot/_main/bazel-out/k8-fastbuild/bin/toolchain/carbon.runfiles/_main/toolchain/install/runtimes/libcxxabi/src/demangle/ItaniumDemangle.h:170:10: fatal error: 
      'ItaniumNodes.def' file not found
  170 | #include "ItaniumNodes.def"
      |          ^~~~~~~~~~~~~~~~~~
1 error generated.
error: failure running `clang` to perform linking: Failed to compile runtime source file 'runtimes/libcxxabi/src/cxa_demangle.cpp'
```

### What did you do, or what's a simple way to reproduce the bug?


`~/tmp/carb/simple.carbon`:
```carbon
import Core library "io";

fn Run() -> i32 {
  Core.Print(1234);
  return 0;
}
```

Run: `./scripts/run_bazelisk.py run //toolchain compile ~/tmp/carb/simple.carbon && ./scripts/run_bazelisk.py run //toolchain -- link --output=/usr/local/google/home/nicholasbishop/tmp/carb/simple ~/tmp/carb/simple.o && ~/tmp/carb/simple`

### What did you expect to happen?


_No response_

### What actually happened?


_No response_

### Any other information, logs, or outputs that you want to share?


_No response_

--- a/toolchain/install/BUILD
+++ b/toolchain/install/BUILD
@@ -342,6 +342,7 @@ filegroup(
         ":libcxx_hdrs",
         ":libcxxabi_hdrs",
         ":libcxxabi_srcs",
+        ":libcxxabi_textual_srcs",
         ":libunwind_hdrs",
         ":libunwind_srcs",
     ],
@@ -728,11 +729,12 @@ filegroup(
 )
 
 py_test(
-    name = "llvm_symlinks_test",
-    size = "small",
-    srcs = ["llvm_symlinks_test.py"],
+    name = "install_test",
+    size = "large",
+    srcs = ["install_test.py"],
     data = [
         ":built_runtimes",
+        ":carbon-busybox",
         ":install_data",
     ],
     deps = ["@bazel_tools//tools/python/runfiles"],

--- a/toolchain/install/install_test.py
+++ b/toolchain/install/install_test.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 
-"""Checks various LLVM tool symlinks behave as expected."""
+"""Integration tests for the Carbon toolchain installation."""
 
 __copyright__ = """
 Part of the Carbon Language project, under the Apache License v2.0 with LLVM
@@ -13,11 +13,12 @@ import subprocess
 import os
 import platform
 import sys
+import textwrap
 import unittest
 from bazel_tools.tools.python.runfiles import runfiles
 
 
-class LLVMSymlinksTest(unittest.TestCase):
+class InstallTest(unittest.TestCase):
     def setUp(self) -> None:
         # The install root is adjacent to the test script
         self.install_root = Path(sys.argv[0]).parent
@@ -117,6 +118,95 @@ class LLVMSymlinksTest(unittest.TestCase):
             raise
         self.assertEqual(run.stderr, "")
         self.assertRegex(run.stdout, r"(^|\n)SUCCESS\n")
+
+    def run_carbon_test(
+        self, name: str, source: str, use_prebuilt: bool, expected_output: str
+    ) -> None:
+        src_file = self.tmpdir / f"{name}.carbon"
+        src_file.write_text(textwrap.dedent(source).lstrip())
+
+        output_bin = self.tmpdir / name
+
+        carbon = self.runfiles.Rlocation(
+            "carbon/toolchain/install/carbon-busybox"
+        )
+
+        try:
+            obj_file = self.tmpdir / f"{name}.o"
+            subprocess.run(
+                [carbon, "compile", f"--output={obj_file}", src_file],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+
+            link_cmd = [carbon]
+            if use_prebuilt:
+                link_cmd.append(f"--prebuilt-runtimes={self.prebuilt_runtimes}")
+            link_cmd.extend(["link", f"--output={output_bin}", obj_file])
+            subprocess.run(link_cmd, check=True, capture_output=True, text=True)
+        except subprocess.CalledProcessError as err:
+            self.fail(f"Subprocess failed: {err.stderr}")
+
+        run = subprocess.run(
+            [output_bin], check=True, capture_output=True, text=True
+        )
+        self.assertEqual(run.returncode, 0)
+        self.assertEqual(run.stdout.strip(), expected_output)
+
+    def run_cpp_test(
+        self, name: str, source: str, use_prebuilt: bool, expected_output: str
+    ) -> None:
+        src_file = self.tmpdir / f"{name}.cpp"
+        src_file.write_text(textwrap.dedent(source).lstrip())
+
+        output_bin = self.tmpdir / name
+
+        clang = self.install_root / "llvm/bin/clang++"
+
+        try:
+            cmd = [clang, f"-o{output_bin}", src_file]
+            if use_prebuilt:
+                cmd.append(
+                    f"-Xcarbon=--prebuilt-runtimes={self.prebuilt_runtimes}"
+                )
+            subprocess.run(cmd, check=True, capture_output=True, text=True)
+        except subprocess.CalledProcessError as err:
+            self.fail(f"Subprocess failed: {err.stderr}")
+
+        run = subprocess.run(
+            [output_bin], check=True, capture_output=True, text=True
+        )
+        self.assertEqual(run.returncode, 0)
+        self.assertEqual(run.stdout.strip(), expected_output)
+
+    def test_carbon_end_to_end(self) -> None:
+        source = r"""
+        import Cpp library "<iostream>";
+        fn Run() -> i32 {
+          Cpp.std.cout << "Hello from Carbon\n";
+          return 0;
+        }
+        """
+        for use_prebuilt in [True, False]:
+            with self.subTest(use_prebuilt=use_prebuilt):
+                self.run_carbon_test(
+                    "simple_carbon", source, use_prebuilt, "Hello from Carbon"
+                )
+
+    def test_cpp_end_to_end(self) -> None:
+        source = r"""
+        #include <iostream>
+        int main() {
+          std::cout << "Hello from C++" << std::endl;
+          return 0;
+        }
+        """
+        for use_prebuilt in [True, False]:
+            with self.subTest(use_prebuilt=use_prebuilt):
+                self.run_cpp_test(
+                    "simple_cpp", source, use_prebuilt, "Hello from C++"
+                )
 
 
 if __name__ == "__main__":

